src/modules/auth/auth.service.ts

```diff
@@ -30,4 +45,31 @@ export class AuthService {
 			refreshToken: this.tokenService.generateRefreshToken(user)
 		};
 	}
+
+	async handleGoogleAuth({ accessToken }: GoogleLoginDto) {
+		try {
+			const googleUserInfo: GoogleUserInfo = await this.fetchGoogleUserProfile(accessToken);
+			const existingUser = await this.userService.handleGetUserByEmail(googleUserInfo.email);
+			// If user is not registered
+			if (!existingUser) return this.userService.handleCreateGoogleUser(googleUserInfo);
+			// If user already exist with such email
+			else return this.login(existingUser);
+		} catch (error) {
+			if (error?.response) throw error;
+			throwHttpException(
+				RESPONSE_TYPES.SERVER_ERROR,
+				`Failed to authenticate with Google : ${error}`
+			);
+		}
+	}
+
+	async fetchGoogleUserProfile(accessToken: string): Promise<GoogleUserInfo> {
+		const googleApiUrl = `${GOOGLE_USER_API_URL}?access_token=${accessToken}`;
+		try {
+			const response = await axios.get(googleApiUrl);
+			return response.data; // Contains email, name, picture, etc.
+		} catch (error) {
+			throwHttpException(RESPONSE_TYPES.SERVER_ERROR, `Failed to fetch user google info`);
```

Comment:
We are not always sure that the error is caused by HTTP failure. I suggest first checking if Axios caused error and handling two scenarios.

```
}catch(error){
    if (error instanceof AxiosError){
        throw new Error("Axios Error: " + error.message);
    }
    throw new Error("Generic Error");
}
```